When a function calls itself, this is called recursion
